bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/soap/SOAPConnector.java

```diff
@@ -57,12 +58,16 @@
  */
 @NonNullByDefault
 public class SOAPConnector {
-    private static final int SOAP_TIMEOUT = 2000; // in ms
+    private static final int SOAP_TIMEOUT = 5; // in s
+    private static final int SOAP_MESSAGE_CACHE_EXPIRY_TIME = 2000; // in ms
     private final Logger logger = LoggerFactory.getLogger(SOAPConnector.class);
     private final HttpClient httpClient;
     private final String endpointBaseURL;
     private final SOAPValueConverter soapValueConverter;
 
+    private final ExpiringCacheMap<SOAPRequest, SOAPMessage> soapMessageCache = new ExpiringCacheMap<>(
```

Comment:
Only a minor remark. Why not passing the time value directly like [here](https://github.com/openhab/openhab-addons/pull/9693/files#diff-5693a957fcf128a6a57fa9f5062b873f08f373818f4f1351bfe01705274af0aaR85) ? Or use `Duration.of()` to improve readability.

Reply:
The stateCache can probably be omitted. Sinc ethe SOAP request to fill it is cached anyway. WDYT?

Reply:
Yes, we probably could remove that one.

Reply:
OTOH, it bypasses the construction of the SOAP request and does not add much complexity now that it is here. So I would prefer to leave it in place.